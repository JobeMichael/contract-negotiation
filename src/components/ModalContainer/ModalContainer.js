import React from "react";
import ReactDOM from "react-dom";
import * as Styled from "./ModalContainer.styles";

const ModalContainer = ({ setOpen, children }) => {
  const close = () => {
    setOpen(false);
  };

  return ReactDOM.createPortal(
    <>
      <Styled.ModalShadow onClick={close} />
      <Styled.Modal>
        <Styled.ModalBanner>
          <a onClick={close}>Close</a>
        </Styled.ModalBanner>
        <Styled.ModalContent>{children}</Styled.ModalContent>
      </Styled.Modal>
    </>,
    document.getElementById("app-modal")
  );
};

export default ModalContainer;
