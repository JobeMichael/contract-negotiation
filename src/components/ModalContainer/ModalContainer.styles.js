import styled from "styled-components";

export const Modal = styled.div`
  min-width: 500px;
  background-color: white;
  position: fixed;
  top: 20%;
  z-index: 5;
  max-height: calc(100% - 200px);
  left: calc(50% - 250px);
  display: flex;
  flex-direction: column;
  @media (max-width: 500px) {
    left: 0px;
    margin: 0px 10px;
  }
  p {
    font-size: 20px;
  }
  a {
    align-self: flex-end;
    color: #004055;
    font-weight: bold;
    cursor: pointer;
  }

  .valid {
    color: green;
  }
  .failure {
    color: red;
  }
`;

export const ModalContent = styled.div`
  overflow: auto;
  min-height: 200px;
  padding: 0px 40px;
  padding-bottom: 80px;
`;

export const ModalFooter = styled.div`
  box-shadow: 0px -2px 10px 0px grey;
  height: 60px;
  display: flex;
  justify-content: center;
`;

export const ModalShadow = styled.div`
  position: fixed;
  height: 100%;
  width: 100%;
  top: 0px;
  background-color: black;
  opacity: 0.7;
  z-index: 4;
`;

export const ModalBanner = styled.div`
  margin-bottom: 20px;
  background-color: #29bdba;
  color: white;
  padding: 10px;
  text-align: right;
`;
