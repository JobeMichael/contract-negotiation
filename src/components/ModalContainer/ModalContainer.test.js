import { act, fireEvent, render } from "@testing-library/react";
import React from "react";
import ReactDOM from "react-dom";
import ModalContainer from "./ModalContainer";

describe("<ModalContainer/>", () => {
  const close = jest.fn();

  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element;
    });
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
    close.mockRestore();
  });

  let renderResult;

  const setup = () => (
    <ModalContainer setOpen={close}>
      <span>Children</span>
    </ModalContainer>
  );

  it("renders Modal properly", () => {
    act(() => {
      renderResult = render(setup());
    });

    expect(renderResult.baseElement).toMatchSnapshot();
  });

  it("calls close button", () => {
    act(() => {
      renderResult = render(setup());
    });

    const closeButton = renderResult.baseElement.querySelector("a");
    fireEvent.click(closeButton);

    expect(close).toBeCalledTimes(1);
  });
});
