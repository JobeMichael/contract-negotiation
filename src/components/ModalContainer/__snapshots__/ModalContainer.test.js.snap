// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalContainer/> renders Modal properly 1`] = `
<body>
  <div>
    <div
      class="sc-AxhCb fSeUtm"
    />
    <div
      class="sc-AxjAm fuyJBj"
    >
      <div
        class="sc-AxhUy jkkAyo"
      >
        <a>
          Close
        </a>
      </div>
      <div
        class="sc-AxirZ kZUdNH"
      >
        <span>
          Children
        </span>
      </div>
    </div>
  </div>
</body>
`;
